File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/cart/CartAdapter.java
package com.example.myapplication.ui.fragment.cart;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapplication.databinding.CardviewCartBinding;

public class CartAdapter extends RecyclerView.Adapter<CartViewHolder> {
    private int cartQuantity ;
    private TextView mTextView;

    CartAdapter() {}

    @NonNull
    @Override
    public CartViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new CartViewHolder(
                CardviewCartBinding.inflate(
                        LayoutInflater.from(parent.getContext()),
                        parent,
                        false
                )
        );
    }

    @Override
    public void onBindViewHolder(final CartViewHolder holder, final int position) {
        holder.bindTo(null);
    }

    @Override
    public int getItemCount() {
        return 1;
//                ShoppingCartItem.getInstance(mContext).getFoodTypeSize();
    }

    public void deleteData(int position) {

    }
}

File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/cart/CartAdapter.kt
package com.example.myapplication.ui.fragment.cart

import android.content.SharedPreferences
import android.view.LayoutInflater
import android.view.ViewGroup
import com.example.myapplication.R
import com.example.myapplication.databinding.CardCartBinding
import com.example.myapplication.model.Cart
import com.example.myapplication.model.Product
import com.example.myapplication.model.total
import com.example.myapplication.util.android.base.BaseAdapter
import com.example.myapplication.util.android.saveCart
import com.google.android.material.textview.MaterialTextView

class CartAdapter(
    items: MutableList<Product>,
    private val sharedPreferences: SharedPreferences,
    private val priceTextView: MaterialTextView
): BaseAdapter<CartViewHolder, Product>(items) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CartViewHolder =
        CartViewHolder(
            CardCartBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        ).apply {
            setMinusClickListener {
                val item = items[adapterPosition]
                if (item.quantity == 1) {
                    items.remove(item)
                    notifyItemRemoved(adapterPosition)
                } else {
                    item.quantity--
                }
                bindTo(item)
                priceTextView.text = it.context.getString(R.string.set_price, items.total())
                saveCart(sharedPreferences, Cart(items))
            }
            setPlusClickListener {
                val item = items[adapterPosition]
                item.quantity++
                bindTo(item)
                priceTextView.text = it.context.getString(R.string.set_price, items.total())
                saveCart(sharedPreferences, Cart(items))
            }
        }
}